lib/puma/cluster.rb

```diff
@@ -178,6 +196,7 @@ def check_workers(force=false)
 
       @workers.delete_if(&:dead?)
 
+      cull_workers
```

Comment:
Maybe this should be rearchitected so that cull_workers and spawn_workers takes an integer argument, and then you calculate the value to give to those methods here. You could then call cull_workers(1) in your TTOU signal trap.